deepfence_worker/ingesters/cloud_resource.go

```diff
@@ -0,0 +1,348 @@
+		}); err != nil {
+		log.Error().Msgf("error: %+v", err)
+		return err
+	}
+
+	return tx.Commit()
+}
+
+func ResourceToMaps(ms []CloudResource) []map[string]interface{} {
+	res := []map[string]interface{}{}
+	for _, v := range ms {
+		res = append(res, v.ToMap())
+	}
+	fmt.Println("check ms size", len(res))
+	return res
+}
+
+func (c *CloudResource) ToMap() map[string]interface{} {
+	out, err := json.Marshal(*c)
+	if err != nil {
+		fmt.Println("check err", err)
+		return nil
+	}
+	bb := map[string]interface{}{}
+	err = json.Unmarshal(out, &bb)
+	if err != nil {
+		fmt.Println("check err 2", err)
+		return nil
+	}
+
+	bb = convertStructFieldToJSONString(bb, "task_definition")
+	bb = convertStructFieldToJSONString(bb, "vpc_options")
+	bb = convertStructFieldToJSONString(bb, "policy")
+	bb = convertStructFieldToJSONString(bb, "public_ips")
+	bb = convertStructFieldToJSONString(bb, "network_interfaces")
+	bb = convertStructFieldToJSONString(bb, "iam_policy")
+	bb = convertStructFieldToJSONString(bb, "ip_configuration")
+	bb = convertStructFieldToJSONString(bb, "security_groups")
+	bb = convertStructFieldToJSONString(bb, "vpc_security_groups")
+	bb = convertStructFieldToJSONString(bb, "container_definitions")
+	bb = convertStructFieldToJSONString(bb, "event_notification_configuration")
+	bb = convertStructFieldToJSONString(bb, "resource_vpc_config")
+	bb = convertStructFieldToJSONString(bb, "network_configuration")
+	bb = convertStructFieldToJSONString(bb, "policy_std")
+	bb = convertStructFieldToJSONString(bb, "attached_policy_arns")
+	bb = convertStructFieldToJSONString(bb, "groups")
+	bb = convertStructFieldToJSONString(bb, "inline_policies")
+	bb = convertStructFieldToJSONString(bb, "instances")
+	bb = convertStructFieldToJSONString(bb, "containers")
+	bb = convertStructFieldToJSONString(bb, "target_health_descriptions")
+	bb = convertStructFieldToJSONString(bb, "instance_profile_arns")
+	bb = convertStructFieldToJSONString(bb, "users")
+	bb = convertStructFieldToJSONString(bb, "user-groups")
+	bb = convertStructFieldToJSONString(bb, "vpc_security_group_ids")
+	bb = convertStructFieldToJSONString(bb, "resources_vpc_config")
+
+	if bb["resource_id"] == "aws_ecs_service" {
+		bb["arn"] = bb["service_name"]
+	}
+	if bb["resource_id"] == "aws_lambda_function" {
+		bb["arn"] = bb["name"]
+	}
+	if bb["resource_id"] == "aws_iam_access_key" {
+		bb["arn"] = bb["access_key_id"]
+	}
+	if strings.Contains("azure", bb["resource_id"].(string)) {
+		bb["arn"] = bb["name"]
+	}
+	if strings.Contains("azure", bb["resource_id"].(string)) {
+		bb["arn"] = bb["name"]
+		if bb["resource_id"].(string) == "azure_compute_virtual_machine" {
+
+			bb["arn"] = bb["vm_id"]
+		}
+	}
+	accountId, present := bb["account_id"]
+	if present {
+		splits := strings.Split(fmt.Sprintf("%v", accountId), "-")
+		if len(splits) > 2 {
+			bb["cloud_provider"] = splits[2]
+		}
+	}
+
+	return bb
+}
+
+// TODO: Call somewhere
+func LinkNodesWithCloudResources(ctx context.Context) error {
```

Comment:
can you can call this at the end of commit ? what will be the impact ? after every commit this has to check and run on large number of data ?

do we need to add some kind of afterfunction for our bulk processor ?